### PR TITLE
Remove controller-manager --service-sync-period flag

### PR DESCRIPTION
**Pull request by khenidak**
**What this PR does / why we need it**:
This PR removes controller manager --service-sync-period flag which is not used anywhere in the code and is causing confusion
 
**Which issue(s) this PR fixes** 
https://github.com/kubernetes/kubernetes/issues/58776

**Special notes for your reviewer**:
@deads2k this remove the flag as per the discussion on #58776 
2 commits 
1. one for code change
2. one for auto generated code

**Release note**:
```release-note
1. Controller-manager --service-sync-period flag is removed (was never used in the code).
```

### CODE REVIEW

**Comment by deads2k** [comment]
This flag was disconnected at least as far back as 1.4.  I'll leave this open for a couple days in case someone can find a way it could have worked in the past year and change.  Lacking that, I plan to merge on Wednesday.

@khenidak please squash into two commits: one for work, one for generated.

/approve

**Comment by lavalamp** [comment]
if anything this flag probably belongs in the cloud controller binary rather than the main one?

File: cmd/kube-controller-manager/app/options/options_test.go [hunk 1]
```diff
@@ -138,7 +138,6 @@ func TestAddFlags(t *testing.T) {
 				EnableProfiling:                                 false,
 				CIDRAllocatorType:                               "CloudAllocator",
 				NodeCIDRMaskSize:                                48,
-				ServiceSyncPeriod:                               metav1.Duration{Duration: 2 * time.Minute},
```
Comment (cheftako) [inline cmd/kube-controller-manager/app/options/options_test.go]: Shouldn't we clean up line 103 as well? ("--service-sync-period=2m",)

**Review by cheftako — APPROVED** [review]
/lgtm. Please make sure to clean up all of cmd/kube-controller-manager/app/options/options_test.go.

File: cmd/kube-controller-manager/app/options/options_test.go (hunk 1 above)
Comment (khenidak) [reply cmd/kube-controller-manager/app/options/options_test.go]: Yes. I am working on flushing out out of tests.

**Comment by khenidak** [comment]
@deads2k squashed -- and all test passed except pull-kubernetes-unit which i believe is a false positive (please let me know if it is not).

**Comment by khenidak** [comment]
/assign @lavalamp 
/assign @mikedanese

Folks, please approve this one.

**Comment by khenidak** [comment]
the e2e is flaky 
`
W0212 20:53:45.956] F0212 20:53:45.956282     777 run_remote.go:233] Could not retrieve list of images based on image prefix "cos-stable-63-10032-71-0": Failed to list images in project "cos-cloud": Get https://www.googleapis.com/compute/beta/projects/cos-cloud/global/images?alt=json: oauth2: cannot fetch token: Post https://accounts.google.com/o/oauth2/token: net/http: TLS handshake timeout
`

I will try later tonight.

**Pull request merged**